File: MultipleActivities/app/src/main/java/com/mhmmdyzci/multipleactivities/MainActivity.java
package com.mhmmdyzci.multipleactivities;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {
    EditText editText;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        editText = findViewById(R.id.editTextTextPersonName);
    }
    public  void changeActivity(View view ){
        String name = editText.getText().toString();
        Intent intent = new Intent(MainActivity.this, MainActivity2.class );
        intent.putExtra("userName", name);

        startActivity(intent);
    }
}

File: KotlinIntent/app/src/main/java/com/mhmmdyzci/kotlinintent/MainActivity.kt
package com.mhmmdyzci.kotlinintent

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import com.mhmmdyzci.kotlinintent.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    //YAŞAM DÖNGÜSÜNE BAK
    private lateinit var binding: ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)
    }
    //Context haberdar yapı
    fun next(view: View){
        val intent = Intent(applicationContext,NextActivity::class.java)
        intent.putExtra("userName",binding.userNameText.text.toString())
        intent.putExtra("name",binding.nameText.text.toString())
        startActivity(intent)
        //önceki aktiviteyi kapatır
        finish()
    }
}